trunner/device.py

```diff
@@ -44,6 +64,110 @@ def run(self, test):
             self.serial.close()
 
 
+class GPIO:
+    def __init__(self, pin):
+        self.pin = pin
+
+        RPi.GPIO.setmode(RPi.GPIO.BCM)
+        RPi.GPIO.setwarnings(False)
+        RPi.GPIO.setup(self.pin, RPi.GPIO.OUT)
+
+    def high(self):
+        RPi.GPIO.output(self.pin, RPi.GPIO.HIGH)
+
+    def low(self):
+        RPi.GPIO.output(self.pin, RPi.GPIO.LOW)
+
+
+class IMXRT106xRunner(DeviceRunner):
+    SDP = 'plo-ram-armv7m7-imxrt106x.sdp'
+    IMAGE = 'phoenix-armv7m7-imxrt106x.disk'
+
+    def __init__(self, port, phoenixd_port='/dev/ttyUSB0'):
+        super().__init__(port)
+        self.phoenixd_port = phoenixd_port
+        self.serial_downloader = False
+
+        self.reset_gpio = GPIO(17)
+        self.reset_gpio.high()
+
+        self.boot_gpio = GPIO(4)
+
+    def reset(self):
+        self.reset_gpio.low()
+        time.sleep(0.050)
+        self.reset_gpio.high()
+
+    def boot(self):
+        if self.serial_downloader:
+            self.boot_gpio.low()
+        else:
+            self.boot_gpio.high()
+
+        self.reset()
+
+    def flash(self):
```

Comment:
I would add some of these functions to the base class `DeviceRunner` and make them no-op in Qemu runners

Reply:
I added class `Runner`.